smart_contract/executor/src/default.rs

```diff
@@ -563,10 +564,42 @@ pub mod account {
 }
 
 pub mod asset_definition {
-    use permission::{account::is_account_owner, asset_definition::is_asset_definition_owner};
+    use permission::{
+        account::is_account_owner, asset_definition::is_asset_definition_owner,
+        domain::is_domain_owner,
+    };
 
     use super::*;
 
+    pub fn visit_register_asset_definition<V: Validate + ?Sized>(
+        executor: &mut V,
+        authority: &AccountId,
+        isi: Register<AssetDefinition>,
+    ) {
+        let domain_id = isi.object.id().domain_id();
+
+        if is_genesis(executor) {
+            pass!(executor);
+        }
+        match is_domain_owner(domain_id, authority) {
+            Err(err) => deny!(executor, err),
+            Ok(true) => pass!(executor),
+            Ok(false) => {}
+        }
+        let can_register_asset_definition_token =
+            tokens::domain::CanRegisterAssetDefinitionInDomain {
+                domain_id: domain_id.clone(),
+            };
+        if can_register_asset_definition_token.is_owned_by(authority) {
+            pass!(executor);
+        }
+
+        deny!(
+            executor,
+            "New `AssetDefinition` can only be registered by domain owner or if granted `CanRegisterAssetDefinitionInDomain` token"
+        );
+    }
+
     pub fn visit_unregister_asset_definition<V: Validate + ?Sized>(
```

Comment:
I think this functions should have a check if `authority` owns the domain

Reply:
AFAIK this check incorporated in `is_asset_definition_owner`